market-generator/src/test/java/com/market/banica/generator/grpc/MarketTestIT.java

```diff
@@ -0,0 +1,235 @@
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+@SpringJUnitConfig
+@SpringBootTest(classes = MarketGeneratorApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("testMarketIT")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class MarketTestIT {
+
+    @Rule
+    public GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+    @Autowired
+    MarketService marketService;
+
+    @Autowired
+    MarketState marketState;
+
+
+    private static final String GOOD = "eggs";
+
+    private String marketName;
+    private ManagedChannel marketChannel;
+
+    private static String marketStateName;
+    private static String marketSnapshotName;
+    private static String marketPropertiesName;
+
+
+    @BeforeAll
+    public static void getFilePath(@Value("${tick.market.state.file.name}") String marketStateFileName,
+                                   @Value("${tick.market.snapshot.file.name}") String marketSnapshotFileName,
+                                   @Value("${market.properties.file.name}") String marketPropertiesFileName) {
+        marketStateName = marketStateFileName;
+        marketSnapshotName = marketSnapshotFileName;
+        marketPropertiesName = marketPropertiesFileName;
+
+    }
+
+    @BeforeEach
+    public void setup() {
+
+        marketName = InProcessServerBuilder.generateName();
+        marketChannel = InProcessChannelBuilder
+                .forName(marketName)
+                .directExecutor()
+                .build();
+
+    }
+
+    @AfterAll
+    public static void cleanUp() throws IOException {
+        ApplicationDirectoryUtil.getConfigFile(marketStateName).delete();
+        ApplicationDirectoryUtil.getConfigFile(marketSnapshotName).delete();
+        ApplicationDirectoryUtil.getConfigFile(marketPropertiesName).delete();
+    }
+
+    @Test
+    public void subscribeForItem_Should_Work() throws InterruptedException, IOException {
```

Comment:
Test condition is missing in the test name.

Reply:
Fixed